src/buffer/out/textBuffer.cpp

```diff
@@ -1957,18 +1956,31 @@ const TextBuffer::TextAndColor TextBuffer::GetText(const bool includeCRLF,
         // retrieve the data from the screen buffer
         auto it = GetCellDataAt(highlight.Origin(), highlight);
 
-        // allocate a string buffer
+        // preallocate to avoid reallocs
+        const auto maxCells = gsl::narrow<size_t>(highlight.Width() + 2); // + 2 for \r\n if we munged it
         std::wstring selectionText;
-        std::vector<COLORREF> selectionFgAttr;
-        std::vector<COLORREF> selectionBkAttr;
+        selectionText.reserve(maxCells);
+        til::small_rle<TextAttribute> attrs{ maxCells, it->TextAttr() };

```

Comment:
FWIW, I am OK with using something memory-inefficient (like a vector with `width` capacity), or even a huge vector with sub-spans, because this is a short-lived transport format rather than a long-lived permanent storage format.

If RLE is slower in this use case, we can spend memory to save time by not using it. 😄

Reply:
On the other hand, I know that til::rle has _very good_ iteration behavior that specifically helps text attribute run production. That is, you can iterate attr-change-by-attr-change... which could be good for HTML/RTF generation.

Reply:
> If RLE is slower in this use case, we can spend memory to save time by not using it

I don't think RLE is any slower than vector type in this case 😅 
 
>  you can iterate attr-change-by-attr-change... which could be good for HTML/RTF generation.

I totally agree. Though, I wasn't planning to make this change in this PR, but since you mentioned it, I'll see if that can be done in this PR itself. I'll add this to the PR checklist for now 👍